### PR TITLE
Core/DB: Update all locales to actual TrinityCore

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
#### Update structure all tables locales
- `locales_achievement_reward` -> `achievement_reward_locale`
- `locales_broadcast_text` -> `broadcast_text_locale`
- `locales_creature_text` -> `creature_text_locale`
- `locales_gossip_menu_option` -> `gossip_menu_option_locale`
- `locales_item` -> `item_template_locale`
- `locales_item_set_names` -> `item_set_names_locale`
- `locales_npc_text` -> `npc_text_locale`
- `locales_page_text` -> `page_text_locale`
- `locales_points_of_interest` -> `points_of_interest_locale`
- `locales_quest` -> `quest_template_locale`

#### Update DB table `trinity string`
- Now correct load data from table

#### Update DB tables to actual TrinityCore
- `npc_text`
- `creature_text`
- `gossip_menu`
- `gossip_menu_option`

#### Delete DB table `db_script_string`
- Now useless

#### Update Broadcast_text system to actual TrinityCore
- Added in `creature_text`
- Added in `gossip_menu_option`
- Added in `npc_text`

## ISSUES ADDRESSED: 
- https://github.com/azerothcore/azerothcore-wotlk/issues/1118
- https://github.com/azerothcore/azerothcore-wotlk/issues/1179

## TESTS PERFORMED:
- [x] Build Windows
- [x] Build in Linux
- [x] Creature text
- [x] Gossip Menu
- [x] Npc text
- [x] Trinity string

## KNOWN ISSUES AND TODO LIST:
- [x] `achievement_reward_locale`
- [x] `broadcast_text_locale`
- [x] `db_script_string`
- [x] `creature_text`
- [x] `creature_text_locale`
- [x] `gossip_menu`
- [x] `gossip_menu_option`
- [x] `gossip_menu_option_locale`
- [x] `item_set_names_locale`
- [x] `item_template_locale`
- [x] `locales_achievement_reward`
- [x] `locales_broadcast_text`
- [x] `locales_creature_text`
- [x] `locales_gossip_menu_option`
- [x] `locales_item`
- [x] `locales_item_set_names`
- [x] `locales_npc_text`
- [x] `locales_page_text`
- [x] `locales_points_of_interest`
- [x] `locales_quest`
- [x] `npc_text`
- [x] `npc_text_locale`
- [x] `page_text_locale`
- [x] `points_of_interest_locale`
- [ ] `quest_greeting` Need implement `ObjectGuid`
- [ ] `quest_greeting_locale` Need implement `quest_greeting`
- [x] `quest_mail_sender`
- [x] `quest_offer_reward_locale`
- [x] `quest_request_items_locale`
- [x] `quest_template_locale`

## P.S.
After startup core appear errors:
```
Ср 02 Января 2019 01:38:55 Loading Gossip menu options...
Ср 02 Января 2019 01:38:55 ERROR: Table `gossip_menu_option` for menu 63, id 2 has unknown option id 20. Option will not be used
Ср 02 Января 2019 01:38:55 ERROR: Table `gossip_menu_option` for menu 64, id 2 has unknown option id 20. Option will not be used
...
```
For this need apply PR https://github.com/azerothcore/azerothcore-wotlk/pull/1218

### List TC commit for this PR
- `achievement_reward` https://github.com/TrinityCore/TrinityCore/commit/95d7a1ff13d2eb23f492501d3042b9a426aae634
- `broadcast_text` https://github.com/TrinityCore/TrinityCore/commit/9a3b28894da14a4d839ddad77fa85be61ff03287
- `creature_text` https://github.com/TrinityCore/TrinityCore/commit/fadc482c37e40da034cca36683402246d1901094
- `gossip_menu` https://github.com/TrinityCore/TrinityCore/commit/5ff8601d36a14e95f9c227db1b85253ff36e551e
- `item_template` and `item_set_names` https://github.com/TrinityCore/TrinityCore/commit/460e2c5dc5eae49f62a9193fb58f1bab42028bd2
- `points_of_interest` and `page_text` https://github.com/TrinityCore/TrinityCore/commit/fbbe247114d092be1d20768419608bfa9cc57608
- `npc_text` https://github.com/TrinityCore/TrinityCore/commit/838b46e33e36eb4b91b86cbfc256eef7e63b4dd5
- `quest` https://github.com/TrinityCore/TrinityCore/commit/3739d8444b1d5683bdac29f4a91df567dc19fd79
- `trinity_string` https://github.com/TrinityCore/TrinityCore/commit/72ed5bfc93af016722c115ef032f7cb1c43d9983 https://github.com/TrinityCore/TrinityCore/commit/27479cf571cf65c9e1657e9c465974449173dc3f
- ` quest_offer_reward_locale` and `quest_request_items_locale` https://github.com/TrinityCore/TrinityCore/commit/2f5403d4af3a90a6e51a377e05a55a0d197afb4d
 - `quest_mail_sender` https://github.com/TrinityCore/TrinityCore/commit/d4e15f1aac3159bb5e17808ee06efb847fdd4789

### CODE REVIEW

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -2195,29 +2196,30 @@ void ObjectMgr::LoadItemLocales()
...
-        {
-            LocaleConstant locale = (LocaleConstant) i;
-            AddLocaleString(fields[1 + 2 * (i - 1)].GetString(), locale, data.Name);
-            AddLocaleString(fields[1 + 2 * (i - 1) + 1].GetString(), locale, data.Description);
-        }
     } while (result->NextRow());
 
-    sLog->outString(">> Loaded %lu Item locale strings in %u ms", (unsigned long)_itemLocaleStore.size(), GetMSTimeDiffToNow(oldMSTime));
-    sLog->outString();
+    sLog->outString(">> Loaded %lu Item Locale strings in %u ms", (uint32)_itemLocaleStore.size(), GetMSTimeDiffToNow(oldMSTime));
```
Comment (Nefertumm) [inline src/server/game/Globals/ObjectMgr.cpp]:
```suggestion
     sLog->outString(">> Loaded %lu Item Locale strings in %u ms", (unsigned long)_itemLocaleStore.size(), GetMSTimeDiffToNow(oldMSTime));
```

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -5212,16 +5210,18 @@ void ObjectMgr::LoadPageTextLocales()
...
+
+        AddLocaleString(Text, locale, data.Text);
 
-        for (uint8 i = 1; i < TOTAL_LOCALES; ++i)
-            AddLocaleString(fields[i].GetString(), LocaleConstant(i), data.Text);
     } while (result->NextRow());
 
-    sLog->outString(">> Loaded %lu PageText locale strings in %u ms", (unsigned long)_pageTextLocaleStore.size(), GetMSTimeDiffToNow(oldMSTime));
-    sLog->outString();
+    sLog->outString(">> Loaded %lu Page Text Locale strings in %u ms", (uint32)_pageTextLocaleStore.size(), GetMSTimeDiffToNow(oldMSTime));
```
Comment (Nefertumm) [inline src/server/game/Globals/ObjectMgr.cpp]:
```suggestion
     sLog->outString(">> Loaded %lu Page Text Locale strings in %u ms", (unsigned long)_pageTextLocaleStore.size(), GetMSTimeDiffToNow(oldMSTime));
```

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan thanks a lot for your contribution.

Can you please confirm that your PR address this issue? https://github.com/azerothcore/azerothcore-wotlk/issues/1179

If there are tables listed in that issue that are NOT addressed by your PR, when you have a minute can you please hightlight which ones? So we can move the remaining changes to a separate issue.

Thank you!

**Comment by Winfidonarleyan** [comment]
@FrancescoBorzi I've updated my PR and think I've answered all your questions

**Comment by FrancescoBorzi** [comment]
Thanks a lot! So this is wip?

**Comment by Winfidonarleyan** [comment]
Yes, WIP

**Comment by BarbzYHOOL** [comment]
@Winfidonarleyan can you please link the TC commit/PR ?

**Comment by Winfidonarleyan** [comment]
@BarbzYHOOL Yes, as there will be time, because a lot of commits

**Comment by Winfidonarleyan** [comment]
@FrancescoBorzi For `quest_greeting` need implement `ObjectGuid` and after added `quest_greeting`.

**Comment by Winfidonarleyan** [comment]
@Nefertumm Yes. https://github.com/TrinityCore/TrinityCore/commit/a64d1ec51eb11fd01a26d08a09de3d8d7dd89675

**Comment by Viste** [comment]
Well done1!!! Great work!!

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan good job! is this ready for testing or you are going to make more changes?

**Comment by Winfidonarleyan** [comment]
@Viste Thanks.
@FrancescoBorzi Yes, Ready to test.

**Comment by FrancescoBorzi** [comment]
@wetbrownsauce can you please have a look at this?

**Comment by Nefertumm** [comment]
Great work, thank you @Winfidonarleyan! Only tests left.

**Comment by wetbrownsauce** [comment]
When running the world update SQL I get the following error:
> SQL Error: Incorrect string value: '\xD0\xAF\xD0\xB6\xD0...' for column 'RewardText' at row 43226

By executing queries piece by piece I've narrowed down the error to this specific line:
```
INSERT INTO `quest_offer_reward_locale` (`ID`, `locale`, `RewardText`, `VerifiedBuild`) SELECT `ID`, `locale`, `OfferRewardText`, `VerifiedBuild` from `quest_template_locale` WHERE `OfferRewardText` != ""; -- Migrate data to new table
```

Any ideas?

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan

**Comment by BarbzYHOOL** [comment]
It's something taken from `quest_template_locale`

**Comment by Winfidonarleyan** [comment]
Try this
```sql
DROP TABLE IF EXISTS `quest_offer_reward_locale`;
CREATE TABLE `quest_offer_reward_locale` (
  `ID` int(10) unsigned NOT NULL DEFAULT '0',
  `locale` varchar(4) NOT NULL,
  `RewardText` text,
  `VerifiedBuild` smallint(6) NOT NULL DEFAULT '0',
  PRIMARY KEY (`ID`,`locale`)
);

DROP TABLE IF EXISTS `quest_request_items_locale`;
CREATE TABLE `quest_request_items_locale` (
  `ID` int(10) unsigned NOT NULL DEFAULT '0',
  `locale` varchar(4) NOT NULL,
  `CompletionText` text,
  `VerifiedBuild` smallint(6) NOT NULL DEFAULT '0',
  PRIMARY KEY (`ID`,`locale`)
);

INSERT INTO `quest_request_items_locale` (`ID`, `locale`, `CompletionText`, `VerifiedBuild`) SELECT `ID`, `locale`, `RequestItemsText`, `VerifiedBuild` from `quest_template_locale` WHERE `RequestItemsText` != ""; -- Migrate data to new table
INSERT INTO `quest_offer_reward_locale` (`ID`, `locale`, `RewardText`, `VerifiedBuild`) SELECT `ID`, `locale`, `OfferRewardText`, `VerifiedBuild` from `quest_template_locale` WHERE `OfferRewardText` != ""; -- Migrate data to new table
```

**Comment by wetbrownsauce** [comment]
I'll try with that in the morning, but when I merged this PR, I dropped all my databases and recreated from scratch using the db_assembler. I've never had an issue until this specific PR.

**Comment by FrancescoBorzi** [comment]
we need to be careful and avoid breaking existing servers

**Comment by wetbrownsauce** [comment]
Still having this problem running the query. Thought it might have something to do with my db application so I tried mysql workbench and still the same issue.

What I did:
1. Dropped auth, characters, world
2. ```CREATE SCHEMA `auth` ;```
3. ```CREATE SCHEMA `characters` ;```
4. ```CREATE SCHEMA `world` ;```
5. Merge PR into source files
6. Run db_assembler to create queries
7. Execute auth_base.sql - no error
8. Execute auth_update.sql - no error
9. Execute characters_base.sql - no error
10. Execute characters_update.sql - no error
11. Execute world_base.sql - no error
12. Execute world_update.sql - error as follows:
err1

I tried running your suggested sql snippet on a fresh world database that doesn't have this PR, still the same error.

**Comment by Winfidonarleyan** [comment]
Now I tried to do it through  `apps/db_assembler/db_assembler.sh` and no errors
```sh
Kargatum@DESKTOP-OQHLEAG MINGW32 /d/Git/azerothcore-wotlk (locales|CHERRY-PICKING)
$ bash apps/db_assembler/db_assembler.sh
1) all: Assemble all
2) bases: Assemble only bases
3) updates: Assemble only updates
4) customs: Assemble only customs
5) import-all: Assemble & Import all
6) import-bases: Assemble & Import only bases
7) import-updates: Assemble & Import only updates
8) import-customs:  Assemble & Import only customs
9) quit: Exit from this menu
[Please enter your choice]: 5
=====     DB ASSEMBLER MENU     =====
===== STARTING ASSEMBLY PROCESS =====
Generating /d/Git/azerothcore-wotlk/env/dist/sql/auth_base ...
Searching on /d/Git/azerothcore-wotlk/data/sql/base/db_auth/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/auth_update ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/db_auth/ ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/pending_db_auth/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/auth_custom ...
Searching on /d/Git/azerothcore-wotlk/data/sql/custom/db_auth/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/characters_base ...
Searching on /d/Git/azerothcore-wotlk/data/sql/base/db_characters ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/characters_update ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/db_characters/ ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/pending_db_characters/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/characters_custom ...
Searching on /d/Git/azerothcore-wotlk/data/sql/custom/db_characters/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/world_base ...
Searching on /d/Git/azerothcore-wotlk/data/sql/base/db_world/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/world_update ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/db_world/ ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/pending_db_world/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/world_custom ...
Searching on /d/Git/azerothcore-wotlk/data/sql/custom/db_world/ ...
=====           DONE            =====
=====       CHECKING DBs        =====
acore_auth database exists
acore_characters database exists
acore_world database exists
=====           DONE            =====
===== STARTING IMPORTING PROCESS =====
acore_auth seems empty
importing AUTH - base ...
acore_characters seems empty
importing CHARACTERS - base ...
acore_world seems empty
importing WORLD - base ...
importing AUTH - update ...
importing CHARACTERS - update ...
importing WORLD - update ...
importing AUTH - custom ...
importing CHARACTERS - custom ...
importing WORLD - custom ...
=====           DONE            =====
1) --skip-column-names
2) bases: Assemble only bases
3) updates: Assemble only updates
4) customs: Assemble only customs
5) import-all: Assemble & Import all
6) import-bases: Assemble & Import only bases
7) import-updates: Assemble & Import only updates
8) import-customs:  Assemble & Import only customs
9) quit: Exit from this menu
[Please enter your choice]: 9
=====     DB ASSEMBLER MENU     =====
Goodbye!

Kargatum@DESKTOP-OQHLEAG MINGW32 /d/Git/azerothcore-wotlk (locales|CHERRY-PICKING)
$
```

**Comment by wetbrownsauce** [comment]
I normally don't import with the assembler, but I'll give that a try and see what happens

**Comment by Winfidonarleyan** [comment]
@wetbrownsauce Usually me too :D. I use from TC DBUpdater with boost

**Comment by wetbrownsauce** [comment]
I normally run all my queries manually through either heidisql or mysql workbench depending on how large they are. I've just been using the default encoding for my DB, what encoding is recommended so I can try with that

**Comment by Winfidonarleyan** [comment]
Utf8

**Comment by wetbrownsauce** [comment]
Yep, that worked, I'm an idiot lol. I'll get to testing now

**Comment by wetbrownsauce** [comment]
Tested. I only know English so that is all I looked into. 

- Went around to a lot of different NPC's, all have correct gossip text and quest text if giving quests.
- Fought some mobs that say things when aggro'd, worked fine.
- Cleared through ICC to see if all text and dialogue matched up.
- Checked with some innkeepers, repair npc's, jeeves, trainers, bankers. All gossip menu options were correct.

**Comment by FrancescoBorzi** [comment]
@Viste @Nefertumm @wetbrownsauce @poszer @BarbzYHOOL @talamortis @Stoabrogga 

sorry for bothering you all, but can you please double check this? it's quite important and I would like to merge it soon (big PRs can easily get merge conflicts)

File: src/server/game/Globals/ObjectMgr.h [hunk 3]
```diff
@@ -469,15 +469,13 @@ typedef UNORDERED_MAP<uint32/*(mapid, spawnMode) pair*/, CellObjectGuidsMap> Map
...
-#define MAX_DB_SCRIPT_STRING_ID        2000010000
 #define MIN_CREATURE_AI_TEXT_STRING_ID (-1)                 // 'creature_ai_texts'
 #define MAX_CREATURE_AI_TEXT_STRING_ID (-1000000)
 
 // Trinity Trainer Reference start range
 #define TRINITY_TRAINER_START_REF      200000
 
-struct TrinityStringLocale
+struct TrinityString
 {
```
Comment (BarbzYHOOL) [inline src/server/game/Globals/ObjectMgr.h]: i hope one day we'll change this xD (no need now)

Comment (Winfidonarleyan) [reply src/server/game/Globals/ObjectMgr.h]: Why?

File: src/server/game/Globals/ObjectMgr.h [hunk 4]
```diff
@@ -1206,15 +1220,17 @@ class ObjectMgr
         }
         GameObjectData& NewGOData(uint32 guid) { return _gameObjectDataStore[guid]; }
         void DeleteGOData(uint32 guid);
-
-        TrinityStringLocale const* GetTrinityStringLocale(int32 entry) const
+       
+		TrinityString const* GetTrinityString(uint32 entry) const
```
Comment (BarbzYHOOL) [inline src/server/game/Globals/ObjectMgr.h]:
```suggestion
	TrinityString const* GetTrinityString(uint32 entry) const
```

**Comment by Winfidonarleyan** [comment]
DB table `trinity_string` is work correctly

**Comment by BarbzYHOOL** [comment]
@wetbrownsauce use the db_assembler all the time, it's awesome :P

Well from the comments it looks like it's good to go. By the way why your DB was not in the right encoding format @wetbrownsauce ? should we add a line about it in the wiki?

@Winfidonarleyan is there a newline at the end of the file? (not sure if needed since we use travis but well)

File: src/server/game/Globals/ObjectMgr.h (hunk 3 above)
Comment (BarbzYHOOL) [reply src/server/game/Globals/ObjectMgr.h]: because we are azeroth

**Comment by BarbzYHOOL** [comment]
And for

```
quest_greeting Need implement ObjectGuid
quest_greeting_locale Need implement quest_greeting
```

Do we have an issue opened for this?

File: src/server/game/Globals/ObjectMgr.h (hunk 3 above)
Comment (FrancescoBorzi) [reply src/server/game/Globals/ObjectMgr.h]: it's not important now

**Comment by Winfidonarleyan** [comment]
For `quest_greeting ` - https://github.com/azerothcore/azerothcore-wotlk/issues/1179
For `ObjectGuid` - No

File: src/server/game/Globals/ObjectMgr.h (hunk 3 above)
Comment (BarbzYHOOL) [reply src/server/game/Globals/ObjectMgr.h]: people with 0 humor there :'(

**Comment by wetbrownsauce** [comment]
@BarbzYHOOL I manually create my DBs and for some reason they were defaulting to latin_swedish_ci. I changed the default now, but I don't think that happens through the assembler

**Comment by Winfidonarleyan** [comment]
https://github.com/azerothcore/azerothcore-wotlk/blob/master/data/sql/create/create_mysql.sql

**Comment by wetbrownsauce** [comment]
Yeah I know, I just had to change my db application's default.

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan is there a newline at the end of the SQL file? (not sure if needed since we use travis but well)

Yes of course

**Comment by Winfidonarleyan** [comment]
I update data to actual TC commit https://github.com/TrinityCore/TrinityCore/commit/d12f0841f198045e80a6bca52c03507df8db5b6d
I also tried using db_assembler `Assembler & Import all`. All successfully
```sh
Kargatum@DESKTOP-OQHLEAG MINGW32 /d/Git/azerothcore-wotlk/apps/db_assembler (locales)
$ bash ./db_assembler.sh
1) all: Assemble all
2) bases: Assemble only bases
3) updates: Assemble only updates
4) customs: Assemble only customs
5) import-all: Assemble & Import all
6) import-bases: Assemble & Import only bases
7) import-updates: Assemble & Import only updates
8) import-customs:  Assemble & Import only customs
9) quit: Exit from this menu
[Please enter your choice]: 5
=====     DB ASSEMBLER MENU     =====
===== STARTING ASSEMBLY PROCESS =====
Generating /d/Git/azerothcore-wotlk/env/dist/sql/auth_base ...
Searching on /d/Git/azerothcore-wotlk/data/sql/base/db_auth/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/auth_update ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/db_auth/ ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/pending_db_auth/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/auth_custom ...
Searching on /d/Git/azerothcore-wotlk/data/sql/custom/db_auth/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/characters_base ...
Searching on /d/Git/azerothcore-wotlk/data/sql/base/db_characters ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/characters_update ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/db_characters/ ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/pending_db_characters/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/characters_custom ...
Searching on /d/Git/azerothcore-wotlk/data/sql/custom/db_characters/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/world_base ...
Searching on /d/Git/azerothcore-wotlk/data/sql/base/db_world/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/world_update ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/db_world/ ...
Searching on /d/Git/azerothcore-wotlk/data/sql/updates/pending_db_world/ ...
Generating /d/Git/azerothcore-wotlk/env/dist/sql/world_custom ...
Searching on /d/Git/azerothcore-wotlk/data/sql/custom/db_world/ ...
=====           DONE            =====
=====       CHECKING DBs        =====
Creating DB acore_auth ...
Creating DB acore_characters ...
Creating DB acore_world ...
=====           DONE            =====
===== STARTING IMPORTING PROCESS =====
acore_auth seems empty
importing AUTH - base ...
acore_characters seems empty
importing CHARACTERS - base ...
acore_world seems empty
importing WORLD - base ...
importing AUTH - update ...
importing CHARACTERS - update ...
importing WORLD - update ...
importing AUTH - custom ...
importing CHARACTERS - custom ...
importing WORLD - custom ...
=====           DONE            =====
1) --skip-column-names
2) bases: Assemble only bases
3) updates: Assemble only updates
4) customs: Assemble only customs
5) import-all: Assemble & Import all
6) import-bases: Assemble & Import only bases
7) import-updates: Assemble & Import only updates
8) import-customs:  Assemble & Import only customs
9) quit: Exit from this menu
[Please enter your choice]: 9
=====     DB ASSEMBLER MENU     =====
Goodbye!

Kargatum@DESKTOP-OQHLEAG MINGW32 /d/Git/azerothcore-wotlk/apps/db_assembler (locales)
```

**Comment by Winfidonarleyan** [comment]
Don't apply this PR.
https://github.com/azerothcore/azerothcore-wotlk/issues/1250

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan do you mean this PR is not ready to be merged?

**Comment by Winfidonarleyan** [comment]
> @Winfidonarleyan do you mean this PR is not ready to be merged?

Ah, yes

**Comment by BarbzYHOOL** [comment]
this pr or #1250?

**Comment by BarbzYHOOL** [comment]
Why are all the tables in myisam?

in your other PR it was like this:

`ENGINE = InnoDB CHARACTER SET = utf8 COLLATE = utf8_general_ci ROW_FORMAT = Compact;`

InnoDB is preferred (myisam is not updated for years). ALso maybe the original issue with encoding was because the setting `COLLATE = utf8_general_ci` wasn't set

worth a try

**Comment by brussens** [comment]
@BarbzYHOOL Well, in general, the correct format, in terms of speed. Myisam works while reading somewhat faster than Innodb engine. Ideally, of course, use Aria engine, but here is a matter of religion (MySQL or mMariaDB).

**Comment by Winfidonarleyan** [comment]
Maybe merge?

**Comment by FrancescoBorzi** [comment]
I tested it again, just compiled, run the server & logged in game. Did a couple of things, talked to NPCs, taking quests, etc... Everything looks normal to me. I have English wow version.

**Comment by BarbzYHOOL** [comment]
thank you winfi, shin and wetbrown!!

**Pull request merged**